src/node/waveshaper.rs

```diff
@@ -308,82 +424,85 @@ impl AudioProcessor for WaveShaperRenderer {
             match self.oversample.load(Ordering::SeqCst).into() {
                 OverSampleType::None => {
                     output.modify_channels(|channel| {
-                        channel.iter_mut().for_each(|o| *o = self.apply_curve(*o));
+                        // curve is always set at this point
+                        let curve = self.curve.as_deref().unwrap();
```

Comment:
This line can be moved after line 423 I think, which will deduplicate it in the 3 match arms.
Also perhaps you could rewrite 423 as a `if let Some(curve) = self.curve {`